docs/specs/authentication-flow-api-reference.md

```diff
@@ -239,6 +266,134 @@ When you are in this step of this flow, you will see a response like the followi
 }
 ```
 
+### Risk assessment and Captcha
+
+Each option may contain the key `risk_assessment` and the key `captcha`.
+
+The shape of `risk_assessment`:
+
+```
+{
+  "required": true,
+  "provider": {
+    "type": "recaptchav3",
+    "alias": "recaptchav3"
+  }
+}
+```
+
+If `risk_assessment.required` is true, then selecting this option requires performing risk assessment first.
+
+You use `provider.type` to determine which client side library to use.
+
+You use `provider.alias` to determine which client side credentials to use.
+
+---
+
+The shape of `captcha`:
+
+```
+{
+  "required": true,
+  "provider": {
+    "type": "recaptchav3",
+    "alias": "recaptchav3"
+  }
+}
+```
+
+If `captcha.required` is true, then selecting this option requires performing captcha challenge first.
+
+You use `provider.type` to determine which client side library to use.
+
+You use `provider.alias` to determine which client side credentials to use.
+
+---
+
+If both are present, you must perform risk assessment first.
+
+#### Risk assessment input
+
+To perform risk assessment, pass an input of the following shape:
+
+```json
+{
+  "risk_assessment": {
+    "alias": "recaptchav3",
+    "type": "recaptchav3",
+    "response": { ... }
```

Comment:
nitpick: How about we use `response_token` for this field?

Reply:
Actually I do not know what `response` we will get from the client side library. My original intention is just pass what the client side library give us there.

Reply:
> I do not know what response we will get from the client side library

Same, but since this field is captcha-provider-specific, I just thought it's safe to use `response_token` for `recaptchav3`, `recaptchav2` & `cloudflare`; We can always use another field name if some non-token response appear

But it's really trivial, can keep response if dun wanna change 🙏

Reply:
I use the name `response` because it is also the actual name the provider use in the API. The phrase "response token" only appears in the description.